### PR TITLE
Skip C overloads

### PR DESCRIPTION
**Pull request by knebekaizer**

### CODE REVIEW

File: Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIrBuilder.kt [hunk 1]
```diff
@@ -146,6 +148,9 @@ class StubsBuildingContextImpl(
 
     private var theCounter = 0
 
+    private val uniqFunctions = mutableSetOf<String>()
+    override fun isOverloading(name: String) = !uniqFunctions.add(name) // TODO params & return type
```
Comment (olonho) [inline Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIrBuilder.kt]: Format  is
``
// TODO: sentence with dot at the end.
``

File: Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIrElementBuilders.kt [hunk 2]
```diff
@@ -412,8 +412,12 @@ internal class EnumStubBuilder(
 
 internal class FunctionStubBuilder(
         override val context: StubsBuildingContext,
-        private val func: FunctionDecl
+        private val func: FunctionDecl,
+        private val skipOverloads: Boolean = false
 ) : StubElementBuilder {
+
+    private fun isOverloading(name: String): Boolean = context.platform == KotlinPlatform.NATIVE && context.isOverloading(name)
```
Comment (olonho) [inline Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIrElementBuilders.kt]: Doesn't it makes impression that we support overloads in JVM case.

File: Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIrBuilder.kt (hunk 1 above)
Comment (sbogolepov) [reply Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIrBuilder.kt]: Why not to pass `FunctionDecl` directly?

File: Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIrElementBuilders.kt [hunk 3]
```diff
@@ -466,6 +470,8 @@ internal class FunctionStubBuilder(
             context.mirror(func.returnType).argType
         }.toStubIrType()
 
+        if (skipOverloads && isOverloading(func.name))
```
Comment (sbogolepov) [inline Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIrElementBuilders.kt]: Do we guarantee that two consequent `cinterop` invocations will return the same result?

Comment (knebekaizer) [reply Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIrElementBuilders.kt]: @sbogolepov  Don't catch the question: nothing is shared between  `cinterop` invocations.
Do you suggest to move `skipOverloads` to `buildingContext` or `InteropConfiguration`?

Comment (sbogolepov) [reply Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIrElementBuilders.kt]: I'm talking about an order of incoming declarations. Is it possible that on the first invocation of `cinterop` the input is `[A, B, C]` and on the second one it is `[B, C, A]`?

Comment (sbogolepov) [reply Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIrElementBuilders.kt]: > Do you suggest to move skipOverloads to buildingContext or InteropConfiguration?
I think it's fine where it is.

**Pull request merged**